src/compiler/program.ts

```diff
@@ -431,13 +431,14 @@ namespace ts {
         return program;
 
         function getCommonSourceDirectory() {
-            if (typeof commonSourceDirectory === "undefined") {
-                if (options.rootDir && checkSourceFilesBelongToPath(files, options.rootDir)) {
+            if (commonSourceDirectory === undefined) {
+                const emittedFiles = filterSourceFilesInDirectory(files, isSourceFileFromExternalLibrary);

```

Comment:
don't think that changes in the way how common subpath is computed should not be the part of this PR since it is not related to original issue